geoAssembler/tests/test_centre_optimiser.py

```diff
@@ -0,0 +1,43 @@
+import geoAssembler.optimiser as centreOptimiser
+import geoAssembler
+
+from extra_data import RunDirectory, stack_detector_data
+from extra_geom import AGIPD_1MGeometry
+
+import numpy as np
+
+import os.path
+
+geom = AGIPD_1MGeometry.from_quad_positions(quad_pos=[
+        (-525, 625),
+        (-550, -10),
+        (520, -160),
+        (542.5, 475),
+    ])
+
+frame_path = os.path.dirname(geoAssembler.__file__) + "/tests/optimiser-test-frame.npy"
+frame = np.load(frame_path)
+
+
+def test_integrator():
+    optimiser = centreOptimiser.CentreOptimiser(geom, frame, sample_dist_m=0.2)
+
+    misaligned_2dint = optimiser.integrate2d(frame).intensity
+    misaligned_2dint_r = optimiser.integrate2d(frame).radial
+    misaligned_2dint_a = optimiser.integrate2d(frame).azimuthal
+
+    assert misaligned_2dint.shape == (957, 832)
+    assert misaligned_2dint_r.shape == (832,)
+    assert misaligned_2dint_a.shape == (957,)
```

Comment:
Asserting on exact numbers like this seems like it's probably quite brittle - if some detail of the implementation changes, it could presumably change by a pixel or two.